File: app/src/main/java/com/nyayadhish/droidgenesiskotlin/lib/utils/AppTextView.java
package com.nyayadhish.droidgenesiskotlin.lib.utils;

import android.annotation.SuppressLint;
import android.content.Context;
import android.util.AttributeSet;
import android.widget.RemoteViews.RemoteView;
import android.widget.TextView;

import androidx.annotation.Nullable;

import org.apache.commons.lang3.StringEscapeUtils;


/**
 * @author Nikhil Nyayadhish
 * This is Custom textview to show emoji.
 * if Imoji not present it will reflect the normal text.
 */

@SuppressLint("AppCompatCustomView")
@RemoteView
public class AppTextView extends TextView {
    public AppTextView(Context context) {
        super(context);
        setText(getText());
    }

    public AppTextView(Context context, @Nullable AttributeSet attrs) {
        super(context, attrs);
        setText(getText());
    }

    public AppTextView(Context context, @Nullable AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        setText(getText());
    }

    public AppTextView(Context context, @Nullable AttributeSet attrs, int defStyleAttr, int defStyleRes) {
        super(context, attrs, defStyleAttr, defStyleRes);
        setText(getText());
    }

    @Override
    public void setText(CharSequence text, BufferType type) {
        if (text != null) {
            super.setText(StringEscapeUtils.unescapeJava(text.toString()), type);
        }
    }
}

File: app/src/main/java/com/nyayadhish/droidgenesiskotlin/lib/utils/AppTextView.kt
package com.nyayadhish.droidgenesiskotlin.lib.utils

import android.annotation.SuppressLint
import android.content.Context
import android.util.AttributeSet
import android.widget.RemoteViews.RemoteView
import android.widget.TextView
import org.apache.commons.lang3.StringEscapeUtils

/**
 * @author Nikhil Nyayadhish
 * This is Custom textview to show emoji.
 * if Imoji not present it will reflect the normal text.
 */
@SuppressLint("AppCompatCustomView")
@RemoteView
class AppTextView : TextView {
    constructor(context: Context?) : super(context) {
        text = text
    }

    constructor(context: Context?, attrs: AttributeSet?) : super(
        context,
        attrs
    ) {
        text = text
    }

    constructor(
        context: Context?,
        attrs: AttributeSet?,
        defStyleAttr: Int
    ) : super(context, attrs, defStyleAttr) {
        text = text
    }

    constructor(
        context: Context?,
        attrs: AttributeSet?,
        defStyleAttr: Int,
        defStyleRes: Int
    ) : super(context, attrs, defStyleAttr, defStyleRes) {
        text = text
    }

    override fun setText(text: CharSequence, type: BufferType) {
        if (text != null) {
            super.setText(StringEscapeUtils.unescapeJava(text.toString()), type)
        }
    }
}